ed25519_utils.go

```diff
@@ -1,20 +1,38 @@
 package jwt
 
 import (
+	"encoding/asn1"
 	"encoding/pem"
 	"golang.org/x/crypto/ed25519"
 )
 
+type OI struct {
+	ObjectIdentifier asn1.ObjectIdentifier
+}
+
+type ed25519PrivKey struct {
```

Comment:
The object-identifier stuff is probably wrong since there is no standard for specifying Ed25519 keys anywhere.